modules/islandora_google_scholar/islandora_google_scholar.module

```diff
@@ -1,282 +1,291 @@
-    if ($degree_grantor) {
-      $tags['citation_dissertation_institution'] = (string) reset($degree_grantor);
-    }
-
-    if ($object['PDF']) {
-      $tags['citation_pdf_url'] = url("islandora/object/$object->id/datastream/PDF/view", array('absolute' => TRUE));
-    }
-
-    $tags['citation_abstract_html_url'] = url("islandora/object/$object->id/", array('absolute' => TRUE));
-  }
-
-  return $tags;
-}
-
-/**
- * Return the year portion of a date.
- *
- * @param string|array $date
- *   The date to parse.  If it is array we use the first element in the array
- *
- * @return null|string
- *   returns the year if the date was parsable and NULL otherwise
- */
-function islandora_google_scholar_parse_date_foryear($date) {
-  if (is_array($date)) {
-    $date = (string) reset($date);
-  }
-  try {
-    // Check if date is currently just a year.
-    if (!(DateTime::createFromFormat('Y', $date))) {
-      $date_obj = new DateTime($date);
-      $date_string = date_format($date_obj, 'Y');
-      return $date_string;
-    }
-    else {
-      return $date;
-    }
-  }
-  catch (Exception $e) {
-    // TODO: should we log this?
-    return NULL;
-  }
-}
-
-/**
- * Adds the meta tags to the HEAD of the html document.
- *
- * @param array $tags
- *   An associate array containing the name => content of the meta tags.
- */
-function islandora_google_scholar_embed_tags($tags) {
-  $weight = 1000;
-  if ($tags != FALSE) {
-    foreach ($tags as $name => $content) {
-      if (is_array($content)) {
-        foreach ($content as $key => $nested_val) {
-          if ($name == 'citation_author') {
-            $weight++;
-          }
-          $element = array(
-            '#tag' => 'meta',
-            '#attributes' => array(
-              'name' => $name,
-              'content' => htmlspecialchars($nested_val),
-            ),
-            '#weight' => $weight,
-          );
-          drupal_add_html_head($element, $key . '_' . $nested_val);
-        }
-      }
-      else {
-        $element = array(
-          '#tag' => 'meta',
-          '#attributes' => array(
-            'name' => $name,
-            'content' => htmlspecialchars($content),
-          ),
-        );
-        drupal_add_html_head($element, $name);
-      }
-    }
-  }
-}
-
-/**
- * Implements hook_CMODEL_islandora_view_object().
- */
-function islandora_google_scholar_ir_citationCModel_islandora_view_object($object) {
-  $tags = islandora_google_scholar_create_meta_tags($object);
-  islandora_google_scholar_embed_tags($tags);
-}
-
-/**
- * Implements hook_CMODEL_islandora_view_object().
- */
-function islandora_google_scholar_ir_thesisCModel_islandora_view_object($object) {
-  $tags = islandora_google_scholar_create_meta_tags($object);
-  islandora_google_scholar_embed_tags($tags);
-}
+<?php
+/**

```

Comment:
@dnwk This is really difficult to review because Git sees this as a complete rewrite of the file, rather than showing just your changes.

I wonder whether putting back the newline at line 2 would get this properly synced?

Reply:
I think I made a mistake in converting EOL to Mac format and causing problems for Github. I could always created a new pull request so that changes will be properly marked. Do you want me to do that?

Reply:
That would be helpful. Thanks.

Reply:
I was able to go back and deleted the troublesome commit. Now Git should properly mark all the changes.